lib/cursor.js

```diff
@@ -56,6 +57,7 @@ class Cursor {
     if (cursorBlock && this.buttons[2] && (!this.lastButtons[2] || cursorChanged)) {
       const vecArray = [new Vec3(0, -1, 0), new Vec3(0, 1, 0), new Vec3(0, 0, -1), new Vec3(0, 0, 1), new Vec3(-1, 0, 0), new Vec3(1, 0, 0)]
       const delta = cursorBlock.intersect.minus(cursorBlock.position)
+      await wait(200)
```

Comment:
no that's not right, there should be a 200ms delay between places, not before the first place

Reply:
ohh I see 1s

Reply:
dont use async/await, it will mess the logic again

Reply:
how would await mess up logic?

Reply:
see https://github.com/PrismarineJS/prismarine-web-client/commit/0506f5a70d6bbae23fde4eddc248b44385f7b8f7

Reply:
As you wrote it, i can already tell you it is already buggy: try spamming right-click very fast: if will call the update multiple time, enter in the if, wait 200ms in parallel and place N blocks at the same time.

Reply:
Changed how it works